R/fwrite.R

```diff
@@ -82,33 +82,40 @@ fwrite = function(x, file="", append=FALSE, quote="auto",
       return(invisible())
     }
   }
-  yaml = if (!yaml) "" else {
+  write_yaml = isTRUE(yaml) || (is.list(yaml) && length(yaml) != 0L)
+  yaml = if (!write_yaml) "" else {
```

Comment:
minor stylistic point -- i always prefer `if (a) X then Y` to `if (!a) Y then X`. i think what you're going for here is being able to see `""` inline but I think code collapse (to `{...}`) is just as good

Reply:
That's just the old code :laughing: 
What I sometimes do is to set the default value unconditionally and then change the default if necessary. In this case it would be something like 
```r
yaml_text = ""
if (yaml) {
 yaml_text = blahblahblah
}
```

Would that work?

Reply:
that's fine w me